mmengine/fileio/backends/oss_backend.py

```diff
@@ -0,0 +1,809 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+import os.path as osp
+import re
+import tempfile
+from contextlib import contextmanager
+from pathlib import Path
+from shutil import SameFileError
+from typing import Generator, Iterator, Optional, Tuple, Union
+
+import mmengine
+from .base import BaseStorageBackend
+
+
+class OSSBackend(BaseStorageBackend):
+    """OSS storage backend for reading (writing) from (to) OSS.
+
+    OSSBackend supports reading and writing data to multiple buckets
+    from Endpoint. If the file path contains the bucket name,
+    OSSBackend will read data from the specified buckets
+    or write data to them. Otherwise, OSSBackend will occur an error.
+
+    Please refer to https://help.aliyun.com/document_detail/85288.html
+    to install the required packages of OSSBackend.
+
+    Args:
+        access_key_id (str, optional): AccessKey ID of your OSS.
+        access_key_secret (str, optional): AccessKey Secret of your OSS.
+        path_mapping (dict, optional): Path mapping dict from local path to
+            ``filepath`` will be replaced by ``dst``. Defaults to None.
+
+    Examples:
+        >>> backend = OSSBackend()
+        >>> filepath1 = 'oss://endpoint/bucket/file'
+        >>> backend.get(filepath1)  # get data from default cluster
+    """
+
+    def __init__(self,
+                 access_key_id: Optional[str] = None,
+                 access_key_secret: Optional[str] = None,
+                 path_mapping: Optional[dict] = None):
```

Comment:
It would be convenient if pass an endpoint as an initialization parameter. Users can use the interface without an endpoint.

```python
from mmengine.fileio import OSSBackend
backend = OSSBackend(
    access_key_id=xxx',
    access_key_secret='xxx',
    endpoint='https://oss-cn-hangzhou.aliyuncs.com')
backend.put_text('hello world!', 'oss://mmengine-test/text1.txt')
# also support the URL contains endpoint
backend.put_text('hello world!', 'oss://oss-cn-hanzhou.aliyuncs.com/mmengine-test/text1.txt')
```

Reply:
Yes，but it may occur that the Endpoint conflicts with folder names, resulting in regular match failures.

Reply:
Can an access_key_id correspond to multiple endpoints? If not, the file path can not contain an endpoint.

Reply:
an `access_key_id` may be for multiple endpoints, so I think that the endpoint needs to be considered.